packages/slice-machine/server/src/api/http/common.ts

```diff
@@ -0,0 +1,52 @@
+import express from "express";
+import getEnv from "../services/getEnv";
+import Environment from "@lib/models/common/Environment";
+import ServerError from "@lib/models/server/ServerError";
+import { TrackerBuilder, Tracker } from "@lib/models/common/tracker";
+
+function stripQuotes(str?: string | undefined): string | undefined {
+  if (!str) return;
+
+  return str.slice(1, str.length - 2);
+}
+
+export interface RequestWithEnv extends express.Request {
+  tracker: Tracker;
+  anonymousId?: string;
+  env: Environment;
+  errors?: { [errorKey: string]: ServerError };
+}
+
+export function WithEnv(
+  handler: (req: RequestWithEnv, res: express.Response) => Promise<any>
+) {
+  return async function (req: express.Request, res: express.Response) {
+    const { env, errors } = await getEnv();
+    const anonymousId = stripQuotes(req.cookies.ajs_anonymous_id);
+    const userId = env.prismicData.userId;
+
+    const tracker = new TrackerBuilder(
+      process.env.SEGMENT_WRITE_KEY || "JfTfmHaATChc4xueS7RcCBsixI71dJIJ",
+      env.updateVersionInfo.currentVersion,
+      {
+        ...(userId ? { userId } : {}),
+        ...(!userId && anonymousId ? { anonymousId } : {}),
+      },
+      env.repo,
+      {
+        flushAt: 20,
+        flushInterval: 300,
+      }
+    ).build();
+
+    const reqWithEnv = (() => {
```

Comment:
I think we mush avoid modifying the request sent to pass some tool to the backend, the request object is not supposed to do that. We can create a context and pass the context to every controller that need it.

Reply:
changing the request is definitely the node approach but instead of making a mutation in a middleware we simply defined a subType of request to explicitly expose more to the handler.

Reply:
With regular handler, you don't get an extra context and using the request to enrich your handler seems to be the current way of doing it in node.